test/Tests/src/Runtime/Asserts_Spec.enso

```diff
@@ -24,4 +25,11 @@ spec = Test.group "Asserts" pending=on_ci <|
         Runtime.assert <|
             4 == 2 + 2
 
+    Test.specify "should fail with type error when given dataflow error as expression" <|
+        p = Panic.catch Assertion_Error (Runtime.assert (Error.throw "foo")) err->
+            err.payload
+        Meta.type_of p . should_be_a Assertion_Error
+        p.message . should_start_with "Result of assert action is a dataflow error"
```

Comment:
Great that we have a test case for it.

Let's maybe add one more case:
```
foo x = Warning.atttach "My warning" (x+2)
assert (foo 2 > 2)
```

----

Currently I checked and it seems that it works all fine, but good to check it since warnings may be error-prone, to avoid regressing this in the future.

Reply:
Good iedea, added in b1b17b22158fdc40bae975407debdfcf1b8433d3